public/controllers/register-agent/core/register-commands/command-generator/command-generator.test.ts

```diff
@@ -0,0 +1,398 @@
+import { CommandGenerator } from './command-generator';
+import {
+  IOSDefinition,
+  IOperationSystem,
+  IOptionalParameters,
+  tOptionalParams,
+} from '../types';
+import { DuplicatedOSException, DuplicatedOSOptionException, NoOSSelectedException, WazuhVersionUndefinedException } from '../exceptions';
+
+const mockedCommandValue = 'mocked command';
+const mockedCommandsResponse = jest.fn().mockReturnValue(mockedCommandValue);
+
+// Defined OS combinations
+export interface ILinuxOSTypes {
+  name: 'linux';
+  architecture: 'x64' | 'x86';
+  extension: 'rpm' | 'deb';
+}
+export interface IWindowsOSTypes {
+  name: 'windows';
+  architecture: 'x86';
+  extension: 'msi';
+}
+
+export interface IMacOSTypes {
+  name: 'mac';
+  architecture: '32/64';
+  extension: 'pkg';
+}
+
+export type tOperatingSystem = ILinuxOSTypes | IMacOSTypes | IWindowsOSTypes;
+
+// Defined Optional Parameters
+
+
+export type tOptionalParameters = 'server_address' | 'agent_name' | 'agent_group' | 'protocol' | 'wazuh_password';
```

Comment:
**suggestion**: if you want to type in the test file, maybe you could use the types defined in `public/controllers/register-agent/config/os-commands-definitions.ts`

Reply:
Yes, but I added new types because if we change the definitions in the os-commands-definitions-ts the tests will fail. And the definition must be considered a configuration that could change.